seldonite/analyze.py

```diff
@@ -59,4 +59,41 @@ def countries(x):
                 df_with_key = df.withColumn("key", lit(key))
 
                 # TODO: what should I return?
-                key.groupBy(col("key")).count().show()
\ No newline at end of file
+                key.groupBy(col("key")).count().show()
+        
+
+        def keywords_over_time_delal(self, year):
+            self.collector._check_args()
+
+            spark_builder = self.collector._get_spark_builder()
+
+            with spark_builder.start_session() as spark_manager:
+                df = self.collector._fetch(spark_manager)
+
+                df = df.withColumn('all_text', psql.functions.concat(df['title'], psql.functions.lit(' '), df['text']))
+
+                year_date = year(col("date")).alias("date_year")
```

Comment:
With the `year` and `month` functions here, I would really recommend doing something like `import pyspark.sql.functions as sfuncs` and using `sfuncs.year` and `sfuncs.month`. It's generally not a good idea stylistically to use wildcard imports because it makes working out where various functions are coming from really confusing